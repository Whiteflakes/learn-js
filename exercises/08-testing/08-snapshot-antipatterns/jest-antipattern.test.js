const { generateDynamic } = require('./index');

test('dynamic snapshot should be stabilized before snapshotting', () => {
  const v = generateDynamic();
  // bad snapshot would include timestamp & random numbers
  // instead stabilize
  v.ts = 12345;
  v.data = v.data.map((x)=>({ i: x.i, n: 0 }));
  v.meta.random = 0;
  v.meta.nested.now = '2020-01-01T00:00:00.000Z';

  expect(v).toMatchSnapshot();
});
