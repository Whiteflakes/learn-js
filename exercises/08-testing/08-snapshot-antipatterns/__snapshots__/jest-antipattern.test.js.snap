// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic snapshot should be stabilized before snapshotting 1`] = `
{
  "data": [
    {
      "i": 0,
      "n": 0,
    },
    {
      "i": 1,
      "n": 0,
    },
    {
      "i": 2,
      "n": 0,
    },
    {
      "i": 3,
      "n": 0,
    },
    {
      "i": 4,
      "n": 0,
    },
    {
      "i": 5,
      "n": 0,
    },
    {
      "i": 6,
      "n": 0,
    },
    {
      "i": 7,
      "n": 0,
    },
    {
      "i": 8,
      "n": 0,
    },
    {
      "i": 9,
      "n": 0,
    },
    {
      "i": 10,
      "n": 0,
    },
    {
      "i": 11,
      "n": 0,
    },
    {
      "i": 12,
      "n": 0,
    },
    {
      "i": 13,
      "n": 0,
    },
    {
      "i": 14,
      "n": 0,
    },
    {
      "i": 15,
      "n": 0,
    },
    {
      "i": 16,
      "n": 0,
    },
    {
      "i": 17,
      "n": 0,
    },
    {
      "i": 18,
      "n": 0,
    },
    {
      "i": 19,
      "n": 0,
    },
    {
      "i": 20,
      "n": 0,
    },
    {
      "i": 21,
      "n": 0,
    },
    {
      "i": 22,
      "n": 0,
    },
    {
      "i": 23,
      "n": 0,
    },
    {
      "i": 24,
      "n": 0,
    },
    {
      "i": 25,
      "n": 0,
    },
    {
      "i": 26,
      "n": 0,
    },
    {
      "i": 27,
      "n": 0,
    },
    {
      "i": 28,
      "n": 0,
    },
    {
      "i": 29,
      "n": 0,
    },
    {
      "i": 30,
      "n": 0,
    },
    {
      "i": 31,
      "n": 0,
    },
    {
      "i": 32,
      "n": 0,
    },
    {
      "i": 33,
      "n": 0,
    },
    {
      "i": 34,
      "n": 0,
    },
    {
      "i": 35,
      "n": 0,
    },
    {
      "i": 36,
      "n": 0,
    },
    {
      "i": 37,
      "n": 0,
    },
    {
      "i": 38,
      "n": 0,
    },
    {
      "i": 39,
      "n": 0,
    },
    {
      "i": 40,
      "n": 0,
    },
    {
      "i": 41,
      "n": 0,
    },
    {
      "i": 42,
      "n": 0,
    },
    {
      "i": 43,
      "n": 0,
    },
    {
      "i": 44,
      "n": 0,
    },
    {
      "i": 45,
      "n": 0,
    },
    {
      "i": 46,
      "n": 0,
    },
    {
      "i": 47,
      "n": 0,
    },
    {
      "i": 48,
      "n": 0,
    },
    {
      "i": 49,
      "n": 0,
    },
  ],
  "meta": {
    "nested": {
      "now": "2020-01-01T00:00:00.000Z",
    },
    "random": 0,
  },
  "ts": 12345,
}
`;
